### PR TITLE
Jgfouca/upstream merge 2020 09 29

### PR DESCRIPTION
**Pull request by jgfouca**
Conflict log:
```
        both modified:   ../../.gitmodules

<<<<<<< HEAD
[submodule "externals/ekat"]
        path = externals/ekat
        url = git@github.com:E3SM-Project/EKAT.git
=======
[submodule "externals/YAKL"]
        path = externals/YAKL
        url = git@github.com:mrnorman/YAKL.git
[submodule "externals/cub"]
        path = externals/cub
        url = git@github.com:NVlabs/cub.git
[submodule "Submod"]
        branch = rljacob/e3smconfig/elm-name
>>>>>>> upstream/master
       RESOLUTION: take both changes

        both modified:   ../../cime_config/allactive/config_compsets.xml

<<<<<<< HEAD
    ATM  = [CAM4, CAM5, SATM, SCREAM]
    LND  = [CLM45,  SLND]
=======
    ATM  = [CAM4, CAM5, SATM]
    LND  = [ELM,  SLND]
>>>>>>> upstream/master
       RESOLUTION: take upstream changes, add SCREAM to upstream ATM list

        both modified:   ../../cime_config/tests.py

<<<<<<< HEAD
    "e3sm_scream" : {
        "time"  : "03:00:00",
        "tests" : (
            "SMS_D.ne4pg2_ne4pg2.F2010-SCREAM-HR",
            "SMS_D.ne4pg2_ne4pg2.F2010-SCREAM-LR",
            "ERS.ne4pg2_ne4pg2.F2010-SCREAM-HR",
            "ERS.ne4pg2_ne4pg2.F2010-SCREAM-LR",
            "ERP.ne4pg2_ne4pg2.F2010-SCREAM-HR.cam-double_memleak_tol",
            "ERP.ne4pg2_ne4pg2.F2010-SCREAM-LR.cam-double_memleak_tol",
            "REP.ne4pg2_ne4pg2.F2010-SCREAM-HR",
            "REP.ne4pg2_ne4pg2.F2010-SCREAM-LR",
            "PEM.ne4pg2_ne4pg2.F2010-SCREAM-HR",
            "PEM.ne4pg2_ne4pg2.F2010-SCREAM-LR",
            )
    }
=======

    "e3sm_gpu" : {
        "tests"    : (
                 "SMS_P36x1_Ld1.T62_oEC60to30v3.CMPASO-NYF",
                 "SMS_P36x1_Ld1.T62_oEC60to30v3.DTESTM",
                 )
    },

>>>>>>> upstream/master
       RESOLUTION: take both changes

        both modified:   ../CMakeLists.txt

<<<<<<< HEAD

  if (DEBUG)
    set(E3SM_DEFAULT_BUILD_TYPE "DEBUG" CACHE STRING "Default build type, inferred from ${DEBUG}")
  else()
    set(E3SM_DEFAULT_BUILD_TYPE "RELEASE" CACHE STRING "Default build type, inferred from ${DEBUG}")
=======
  if (USE_CUDA)
    set(USE_CUDA TRUE PARENT_SCOPE)
  else()
    set(USE_CUDA FALSE PARENT_SCOPE)
>>>>>>> upstream/master
       RESOLUTION: take both changes

<<<<<<< HEAD
# We do want CMAKE_BUILD_TYPE to be set, but we do NOT want CMake to
# decide what optimization flags to append, based on build type,
# so make all the following empty
set (CMAKE_C_FLAGS_RELEASE "")
set (CMAKE_CXX_FLAGS_RELEASE "")
set (CMAKE_Fortran_FLAGS_RELEASE "")

set (CMAKE_C_FLAGS_DEBUG "")
set (CMAKE_CXX_FLAGS_DEBUG "")
set (CMAKE_Fortran_FLAGS_DEBUG "")
=======
if(USE_CUDA)
  enable_language(CUDA)
endif()
>>>>>>> upstream/master
       RESOLUTION: take both changes

        both modified:   ../cam/bld/namelist_files/namelist_defaults_cam.xml

<<<<<<< HEAD
<ncdata dyn="se" hgrid="ne30np4"  nlev="128"            ic_ymd="101" >atm/cam/inic/homme/cami_mam3_Linoz_ne30np4_SCREAM_L128_c160214.nc</ncdata>
=======
<ncdata dyn="se" hgrid="ne45np4"  nlev="72"             ic_ymd="101" >atm/cam/inic/homme/cami_mam3_Linoz_ne45np4_L72_c20200611.nc</ncdata>
>>>>>>> upstream/master
       RESOLUTION: take both changes

<<<<<<< HEAD
<srf_flux_avg eddy_scheme="SHOC_SGS">0</srf_flux_avg>
<srf_flux_avg use_MMF="1"          >1</srf_flux_avg>
=======
<srf_flux_avg use_MMF="1"          >0</srf_flux_avg> !!!!!!!!!
>>>>>>> upstream/master
       RESOLUTION: take SCREAM changes! (NEEDS REVIEW)

        both modified:   ../cam/src/physics/cam/physpkg.F90

<<<<<<< HEAD
       if (do_clubb_sgs .or. do_shoc_sgs) then
          sh_e_ed_ratio = 0.0_r8
       endif
=======
         !======================================================================
         ! Aerosol wet chemistry determines scavenging and transformations.
         ! This is followed by convective transport of all trace species except
         ! water vapor and condensate. Scavenging needs to occur prior to
          ! transport in order to determine interstitial fraction.
         !======================================================================
>>>>>>> upstream/master
       RESOLUTION: take both changes

        added by us:     ../elm/src/external_models/mpp

       RESOLUTION: add the submodule

        added by us:     ../elm/src/external_models/sbetr

       RESOLUTION: add the submodule
```

### CODE REVIEW

File: components/cam/src/physics/cam/physpkg.F90 [hunk 1]
```diff
@@ -2687,70 +2650,44 @@ subroutine tphysbc (ztodt,               &
...
-       !     to determine the interstitial fraction) 
-       !===================================================
+         call t_startf('tphysbc_aerosols')
 
-       call t_startf('bc_aerosols')
-       if (clim_modal_aero .and. .not. prog_modal_aero) then
-          call modal_aero_calcsize_diag(state, pbuf)
-          call modal_aero_wateruptake_dr(state, pbuf)
-       endif
+         if (do_clubb_sgs) sh_e_ed_ratio = 0.0_r8
```
Comment (bogensch) [inline components/cam/src/physics/cam/physpkg.F90:2669]: redundant specification.  Can move lines 2663-2667 down here to replace this.

**Review by bogensch — APPROVED** [review]

**Comment by jgfouca** [comment]
Due to all the submodule churn, the AT is unlikely to be able to handle this PR.

**Comment by jgfouca** [comment]
@brhillman ^

**Comment by jgfouca** [comment]
@bogensch , @brhillman , plz see latest commit.

I was unsure how to resolve this conflict:

```
        both modified:   ../cam/bld/namelist_files/namelist_defaults_cam.xml
<<<<<<< HEAD
<srf_flux_avg eddy_scheme="SHOC_SGS">0</srf_flux_avg>
<srf_flux_avg use_MMF="1"          >1</srf_flux_avg>
=======
<srf_flux_avg use_MMF="1"          >0</srf_flux_avg> !!!!!!!!!
>>>>>>> upstream/master
```

My initial resolution was to keep our (scream's) changes, so the resolution was:

```
<srf_flux_avg eddy_scheme="SHOC_SGS">0</srf_flux_avg>
<srf_flux_avg use_MMF="1"          >1</srf_flux_avg>
```

@bogensch approved the changes above, but I don't know if he looked at this piece closely. After talking to @brhillman , he thinks it should be:

```
<srf_flux_avg eddy_scheme="SHOC_SGS">0</srf_flux_avg>
<srf_flux_avg use_MMF="1"          >0</srf_flux_avg>
```

Please advise.

**Review by brhillman — APPROVED** [review]

**Comment by bogensch** [comment]
Sorry, I missed that. I defer to @brhillman on what the MMF setting should be for this.  The SHOC setting looks good to me.

**Comment by jgfouca** [comment]
Merging manually. Before I added the missing submodules, we did get a passing test run. With the submodules added, the AT (Git, specifically) is not smart enough to do the upstream merge that is part of an AT run.

**Pull request merged**